app/views/collection_items/_item_fields.html.erb

```diff
@@ -7,9 +7,13 @@
   <%= fields_for("collection_items[]", collection_item) do |form| %>
     <!--title-->
     <h4 class="heading" id="collection_item_<%= collection_item.id %>">
-      <%= error_messages_for collection_item %>    
+      <%= error_messages_for collection_item %>
       <%= form.hidden_field :collection_id %>
-      <%= link_to(item_name, collection_item.item) %>   
+      <% if collection_item.item_type == "Work" && collection_item.item.in_unrevealed_collection? && !collection_item.collection.unrevealed? %>
```

Comment:
This perfectly illustrates why I don't touch the challenge code. The logic makes my head hurt, let alone the code :smile: